reactor-core/src/main/java/reactor/core/publisher/MonoToCompletableFuture.java

```diff
@@ -27,6 +27,11 @@
 final class MonoToCompletableFuture<T> extends CompletableFuture<T> implements CoreSubscriber<T> {
 
 	final AtomicReference<Subscription> ref = new AtomicReference<>();
+	final boolean cancelSourceOnNext;
+
+	MonoToCompletableFuture(boolean sourceCanEmitMoreThanOnce) {
```

Comment:
I can't find any use of this constructor where `sourceCanEmitMoreThanOnce` is `true`. Do we even need it?

Reply:
no we don't use it but since this is a `Subscriber` nothing in the constructor would force us to use it exclusively with a `Mono`. If we ever use it on a `Flux` that signature will force us to consider the cancel-in-onNext case (which would become necessary) rather than totally forget about it.